extensions/data-transfer/portability-data-transfer-facebook/src/test/java/org/datatransferproject/transfer/facebook/photos/FacebookPhotosExporterTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.transfer.facebook.photos;
+
+import com.google.common.base.Preconditions;
+import com.restfb.types.Album;
+import com.restfb.types.Photo;
+import org.datatransferproject.spi.transfer.provider.ExportResult;
+import org.datatransferproject.types.transfer.auth.AppCredentials;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.datatransferproject.types.transfer.models.photos.PhotoAlbum;
+import org.datatransferproject.types.transfer.models.photos.PhotoModel;
+import org.datatransferproject.types.transfer.models.photos.PhotosContainerResource;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+
+public class FacebookPhotosExporterTest {
+  private static final String ALBUM_NAME = "My Album";
+  private static final String ALBUM_ID = "1946595";
+  private static final String ALBUM_DESCRIPTION = "This is a test album";
+
+  private static final String PHOTO_ID = "937644721";
+  private static final String PHOTO_SOURCE = "https://www.example.com/photo.jpg";
+  private static final String PHOTO_NAME = "Example photo";
+
+  private FacebookPhotosExporter facebookPhotosExporter;
+  private UUID uuid = UUID.randomUUID();
+
+  @Before
+  public void setUp() {
+    FacebookPhotosInterface photosInterface =
+        new FacebookPhotosInterface() {
+          @Override
+          public Iterable<List<Album>> getAlbums() {
```

Comment:
We have used mockito for configuring this kind of behavior. Here is an example in case you find it useful if this gets unwieldy 

https://github.com/google/data-transfer-project/blob/8fbeadf3ef7ac47d489abac1c9862d300d9a89b6/extensions/data-transfer/portability-data-transfer-flickr/src/test/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporterTest.java#L111

Reply:
Makes sense, switched to Mockito.